types/underscore/underscore-tests.ts

```diff
@@ -486,6 +486,63 @@ _.chain(anyValue)
     .find(i => i.findBooleanFunction())
     .value();
 
+// $ExpectType { valueProperty: string; } | undefined
+_.chain([
+    {
+        group: 'a',
+        subGroup: 1,
+        value: { valueProperty: 'first' }
+    },
+    {
+        group: 'b',
+        subGroup: 2,
+        value: { valueProperty: 'second' }
+    },
+    {
+        group: 'b',
+        subGroup: 3,
+        value: { valueProperty: 'third' }
+    }])
+    .groupBy(i => i.group)
+    .filter(i => i.length >= 2)
```

Comment:
I decided that it would probably be nice to add one more test of this style that covers some earlier updated functions and ties a few functions of the current set into it. When putting this together, I had originally planned to use `.where({ length: 2 })` here instead of `.filter(i => i.length >= 2)`, but I ran into an interesting behavior of `Partial` around arrays that got in the way. Instead of making all of the properties on an array optional, `Partial<SomeType[]>` actually changes the type to `(SomeType | undefined)[]` and still requires every other property including functions to be supplied, which makes a constraint like `{ length: 2 }` not work correctly.

I think this is an edge case that's probably very rarely used and that can be easily worked around by using an `any` assertion, but I thought I'd run it by you quick to see what your thoughts were @jgonggrijp since I could either switch to something other than `Partial` or use a conditional type that handles `any[]` specially to deal with this sooner rather than later if you think it's worth it.

Reply:
Arguably, this is a design mistake in the TS standard library (why can they never just be consistent at M$?!?). On the one hand, using a conditional type that treats `any[]` specially will allow for better type inference. On the other hand, doing so amounts to working around a bug in TS, which feels a bit ugly.

If I were the one doing the work, I might choose the conditional type, but I wouldn't blame you for leaving it as is.

Reply:
Interestingly this is something that applies very specifically to arrays exactly; if I create an interface that extends `Array` and adds properties, then I end up with the non-special behavior again.

```TS
interface ExtendedArray<T> extends Array<T> {
    notAnArrayProperty: string;
}

const partialExtendedArray: Partial<ExtendedArray<string>>; // normal partial type that includes an optional `notAnArrayProperty` and also updates the type of all items to `string | undefined`
```

Extending `Array` is admittedly probably not something that folks do often, but the difficulty I'm running into in trying to correctly handle the extended array case in the implementation for dealing with the non-extended array case makes me inclined to leave this as-is since I don't expect this to be a super common case in general and using `filter` with a function iteratee is an available workaround for this issue.